### PR TITLE
feat: add code implementation of the basic framework of resource I/O …

### LINKED ISSUE #20: Add a feature module called 'kosmos-agent'

### PR DESCRIPTION
**Pull request by yuleichun-striving**
What type of PR is this?
feature

What does this PR do?
add code implementation of the basic framework of resource I/O isolation

Which issue(s) does this PR fix?
Fixes https://github.com/kosmos-io/kosmos/issues/20

Special notes for your reviewer:
Review whether the code implementation is reasonable

Related Projects:
[kosmos-io](https://github.com/kosmos-io/kosmos)

### CODE REVIEW

**Comment by duanmengkk** [comment]
we can consider if  to use controller-runtime to  unify the informer framework

File: cmd/koslet/app/koslet.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		return err
+	}
+
+	podInformerFactory.Start(ctx.Done())
+	podInformerFactory.WaitForCacheSync(ctx.Done())
+	newKoslet.Run()
+	return nil
+}
+
+func buildClient() (kubernetes.Interface, error) {
```
Comment (wuyingjun-lucky) [inline cmd/koslet/app/koslet.go]: Can we unify the clientset building process or we can put it into a common utils

Comment (yuleichun-striving) [reply cmd/koslet/app/koslet.go]: Okay, I'll modify it

**Pull request closed without merging**